flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceWithWatermarkExtractor.java

```diff
@@ -95,8 +97,7 @@ public class TestIcebergSourceWithWatermarkExtractor implements Serializable {
               .setRpcServiceSharing(RpcServiceSharing.DEDICATED)
               .setConfiguration(
                   reporter.addToConfiguration(
-                      // disable classloader check as Avro may cache class in the serializers.
-                      new Configuration().set(CoreOptions.CHECK_LEAKED_CLASSLOADER, false)))
+                      new Configuration().set(PipelineOptions.ALLOW_UNALIGNED_SOURCE_SPLITS, true)))
```

Comment:
we probably need both options here

Reply:
I run the test 1000 times without the classloader issue to appear. Any idea how to check? Since the same issue appears with the configuration set, I suspect that the error message is a red herring

Reply:
> I run the test 1000 times without the classloader issue to appear. Any idea how to check?

not really. in the past, this tends to happen during CI build.

I will leave the decision to you. if it happen again, we would need another PR.